### PR TITLE
Add function to search the associated element of a single point 

### PR DESCRIPTION
**Pull request by zhangchonglin**
- Add function to search the associated element of a single point/particle;
- Delete not needed code segment within the `search_mesh_2d` function.

### CODE REVIEW

File: src/pumipic_adjacency.hpp [hunk 1]
```diff
@@ -1141,5 +1116,97 @@ bool search_mesh_2d(o::Mesh& mesh, // (in) mesh
   return found;
 }
 
+OMEGA_H_DEVICE o::LO search_mesh_2d(const o::Read<o::I8> side_is_exposed,
```
Comment (cwsmith) [inline src/pumipic_adjacency.hpp:1119]: Should the name be changed to make it clearer that this is for a single particle?

**Review by cwsmith — COMMENTED** [review]
Looks good to me.

File: src/pumipic_adjacency.hpp (hunk 1 above)
Comment (zhangchonglin) [reply src/pumipic_adjacency.hpp:1119]: At the time, I was hesitate to introduce new function name, so just kept the same.

Comment (diamog) [reply src/pumipic_adjacency.hpp:1119]: I agree about changing the name. Adding `_pt` at the end is good enough to me. 

Using the same name is a good way to confuse me in the future when working on the search functions.

Comment (zhangchonglin) [reply src/pumipic_adjacency.hpp]: I agree it's better to use a different name to avoid future confusion.

**Pull request merged**